tests/integration/test_persistence.py

```diff
@@ -0,0 +1,92 @@
+#!/usr/bin/env python3
+# Copyright 2022 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+import logging
+from datetime import datetime, timedelta, timezone
+from pathlib import Path
+
+import pytest
+import yaml
+from helpers import get_unit_address, is_alertmanager_up
+from pytest_operator.plugin import OpsTest
+
+from alertmanager_client import Alertmanager
+
+logger = logging.getLogger(__name__)
+
+METADATA = yaml.safe_load(Path("./metadata.yaml").read_text())
+app_name = METADATA["name"]
+resources = {"alertmanager-image": METADATA["resources"]["alertmanager-image"]["upstream-source"]}
+
+
+@pytest.mark.abort_on_fail
+async def test_silences_persist_across_upgrades(ops_test: OpsTest, charm_under_test, httpserver):
+    # deploy alertmanager charm from charmhub
+    logger.info("deploy charm from charmhub")
+    await ops_test.model.deploy("ch:alertmanager-k8s", application_name=app_name, channel="edge")
+    await ops_test.model.wait_for_idle(apps=[app_name], status="active", timeout=1000)
+
+    # set a silencer for an alert and check it is set
+    alertmanager = Alertmanager(address=await get_unit_address(ops_test, app_name, 0))
+
+    silence_start = datetime.now(timezone.utc)
+    silence_end = silence_start + timedelta(minutes=30)
+    matchers = [
+        {
+            "name": "alertname",
+            "value": "fake-alert",
+            "isRegex": False,
+        }
+    ]
+    alertmanager.set_silences(matchers, silence_start, silence_end)
+    silences = alertmanager.get_silences()
+    assert len(silences)
+
+    # Use kubectl to send a SIGTERM signal to Alertmanager so that data is flushed to disk.
+    # This step should not be necessary once the bug in the following issue ticket is fixed
```

Comment:
```suggestion
    # This step should not be necessary once Juju 2.9.33 is released.
```

Reply:
There is no guarantee that 2.9.33 is the release it will be fixed in, it could be earlier or later.